Firestore/src/BulkWriter.php

```diff
@@ -775,6 +889,17 @@ private function createWritesBatchIds($waitForRetryableFailures)
      */
     private function sendBatch(array $writes, array $options = [])
     {
+        $rateLimiterDelayMs = $this->rateLimiter->getNextRequestDelayMs(count($writes));
+        // avoid very long sleep
+        $rateLimiterDelayMs = min(
+            $rateLimiterDelayMs,
+            self::DEFAULT_BACKOFF_MAX_DELAY_MS
+        );
+        if ($rateLimiterDelayMs > 0) {
+            usleep($rateLimiterDelayMs * 1000);
+        }
+        $this->rateLimiter->tryMakeRequest(count($writes));
```

Comment:
This method returns a boolean, but doesn't appear to utilize it. We can touch base on this offline.

Reply:
Yes, its used only for testing the `Ratelimiter`.